File: algoExpert/Arrays/Easy/isValidSubsequence.cpp
#include <vector>
using namespace std;

bool isValidSubsequence(vector<int> array, vector<int> sequence) {
  // Write your code here.
  int sequenceStart {};
  for(int i = 0; i < array.size(); i++){
    int current = array[i];
    if(current == sequence[sequenceStart]){
      sequenceStart++;
    }
  }
  return sequenceStart == sequence.size();
}

File: algoExpert/Arrays/Easy/isValidSubsequence.py
def isValidSubsequence(array, sequence):
    # Write your code here.
    sequenceI = 0
    for i in range(len(array)):
        currentNum = array[i]
        if currentNum == sequence[sequenceI]:
            sequenceI += 1
        if sequenceI == len(sequence):
            return True
    return sequenceI == len(sequence)
